src/extension.ts

```diff
@@ -221,4 +221,130 @@ export function activate(
+                value: number
+                min?: number
+                max?: number
+                low?: number
+                high?: number
+                optimum?: number
+                hoverMessage?: string
+            }
+        }
+
+        context.subscriptions.add(
+            (sourcegraph as any).app.registerFileDecorationProvider({
+                provideFileDecorations: async ({
+                    files,
+                    uri,
+                }: {
+                    uri: string
+                    files: {
+                        uri: string
+                        isDirectory: boolean
+                        name: string
+                        path: string
+                    }[]
+                }) => {
+                    const { repo, rev } = resolveDocumentURI(uri)
+                    const apiParams = codecovParamsForRepositoryCommit({ repo, rev }, sourcegraph)
+
+                    // Fetch commit coverage to get ratio for files, tree coverage to get ratio for directories
+                    const nonDirectoryFiles = files.filter(file => !file.isDirectory)
+                    const directories = files.filter(file => file.isDirectory)
+                    const [commitCoverage, ...treeCoverages] = await Promise.allSettled([
+                        getCommitCoverage(apiParams),
+                        ...directories.map(async (dir, i) => {
+                            const treeCoverage = await getTreeCoverage({ ...apiParams, path: dir.path })
+                            return { treeCoverage, directory: directories[i] }
+                        }),
+                    ])
+
+                    // Iterate over files and get value from commit coverage to construct file decorations
+                    const fileDecorations: FileDecoration[] = []
+
+                    if (commitCoverage.status === 'fulfilled' && commitCoverage.value) {
+                        const { files: reportFiles } = commitCoverage.value.commit.report
+                        for (const file of nonDirectoryFiles) {
+                            const report = reportFiles[file.path]
+                            if (!report) {
+                                continue
+                            }
+
+                            const ratio = parseInt(report.t.c, 10)
+
+                            fileDecorations.push({
+                                uri: file.uri,
+                                after: {
+                                    contentText: `${ratio}%`,
+                                    color: 'gray',
+                                    hoverMessage: `the coverage ratio is ${ratio}%`,
+                                },
+                                meter: {
+                                    value: ratio,
+                                    hoverMessage: `the coverage ratio is ${ratio}%`,
+
+                                    min: 0,
+                                    low: 25,
```

Comment:
Codecov's own logic for their badges and meters seems to be very strict, with everything below 70% already being red: https://docs.codecov.io/docs/coverage-configuration (i.e. `low: 70`)
Should we mirror that?